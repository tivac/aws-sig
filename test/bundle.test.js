"use strict";

const { transformSync } = require("@babel/core");
const { rollup } = require("rollup");
const requireFromString = require("require-from-string");

const env = require("../build/rollup-environment.js");

const config = require("./lib/config.js");

describe("bundle test", () => {
    // const { node : ver } = process.versions;
    // const [ major ] = ver.split(".");

    // // Node < 10 doesn't support new URL()
    // const fn = parseInt(major, 10) >= 10 ? it : it.skip;

    it("should sign requests", async () => {
        const bundle = await rollup({
            input : "./src/index.js",

            plugins : [
                env,
            ],
        });

        const { code } = await bundle.generate({
            format : "cjs",
        });

        const transformed = transformSync(code);

        const sign = requireFromString(transformed);

        expect(
            sign({
                url     : "https://aws.amazon.com/foo",
                headers : {
                    "X-Amz-Date" : "20150830T123600Z",
                },
            }, config())
        ).toMatchSnapshot();
    });
});
