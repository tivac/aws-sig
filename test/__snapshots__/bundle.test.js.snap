// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bundle test should sign requests 1`] = `
Object {
  "headers": Object {
    "Authorization": "AWS4-HMAC-SHA256 Credential=AKIDEXAMPLE/20150830/us-east-1/service/aws4_request, SignedHeaders=x-amz-date, Signature=89495e6980bd197c7dc79990923c6f64d140f76015352de399a981b6990a2c13",
    "X-Amz-Date": "20150830T123600Z",
  },
  "test": Object {
    "auth": "AWS4-HMAC-SHA256 Credential=AKIDEXAMPLE/20150830/us-east-1/service/aws4_request, SignedHeaders=x-amz-date, Signature=89495e6980bd197c7dc79990923c6f64d140f76015352de399a981b6990a2c13",
    "canonical": "GET
/foo

x-amz-date:20150830T123600Z

x-amz-date
e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855",
    "sts": "AWS4-HMAC-SHA256
20150830T123600Z
20150830/us-east-1/service/aws4_request
0f20062628b8dd2af63905b61f3af43e7007f615421ccbdd0a9a440fa0befeae",
  },
  "url": "https://aws.amazon.com/foo",
}
`;
